Loki cloud integration instructions (and necessary mixin changes) (#8492)
grafana/loki: docs/sources/installation/helm/monitor-and-alert/with-grafana-cloud.md

---
title: Configure Monitoring and Alerting of Loki Using Grafana Cloud
menuTitle: Monitor Loki with Grafana Cloud
description: setup monitoring and alerts for Loki using Grafana Cloud
aliases:
  - /docs/installation/helm/monitoring/with-grafana-cloud
weight: 100
keywords:
  - monitoring
  - alert
  - alerting
  - grafana cloud
---

# Configure Monitoring and Alerting of Loki Using Grafana Cloud

> This guide will walk you through using Grafana Cloud to monitor a Loki installation that is installed with the Helm chart. This approach leverages many of the chart's _self monitoring_ features, but instead of sending logs back to Loki itself, it sends them to a Grafana Cloud Logs instance. This approach also does not require the installation of the Prometheus Operator and instead sends metrics to a Grafana Cloud Metrics instance. Using Grafana Cloud to monitor Loki as the added benefit of being able to troubleshoot problems with Loki when the Helm installed Loki is down, as the logs will still be available in the Grafana Cloud Logs instance.

Reviewer: ```suggestion
This guide will walk you through using Grafana Cloud to monitor a Loki installation that is installed with the Helm chart. This approach leverages many of the chart's _self monitoring_ features, but instead of sending logs back to Loki itself, it sends them to a Grafana Cloud Logs instance. This approach also does not require the installation of the Prometheus Operator and instead sends metrics to a Grafana Cloud Metrics instance. Using Grafana Cloud to monitor Loki has the added benefit of being able to troubleshoot problems with Loki when the Helm installed Loki is down, as the logs will still be available in the Grafana Cloud Logs instance.
```

> This guide will walk you through using Grafana Cloud to monitor a Loki installation that is installed with the Helm chart. This approach leverages many of the chart's _self monitoring_ features, but instead of sending logs back to Loki itself, it sends them to a Grafana Cloud Logs instance. This approach also does not require the installation of the Prometheus Operator and instead sends metrics to a Grafana Cloud Metrics instance. Using Grafana Cloud to monitor Loki as the added benefit of being able to troubleshoot problems with Loki when the Helm installed Loki is down, as the logs will still be available in the Grafana Cloud Logs instance.

Author: fixed

**Before you begin:**

- Helm 3 or above. See [Installing Helm](https://helm.sh/docs/intro/install/).

> - A Grafana Cloud account and instance

Reviewer: (nit) Isn't it called a stack instead of an instance? https://grafana.com/docs/grafana-cloud/account-management/cloud-portal/#your-grafana-cloud-stack

Reviewer 2: This would make sense if that stack includes Grafana Cloud Metrics (since line 17 says this will send metrics to a Grafana Cloud Metrics instance).  Otherwise shouldn't we also list GFM as a prerequisite?

Author: I will change to stack, as metrics is required.

- [Grafana Kubernetes Monitoring using Agent](/docs/grafana-cloud/kubernetes-monitoring/configuration/config-k8s-agent-guide/) configured for the Kubernetes cluster
- A running Loki deployment installed in that Kubernetes cluster via the Helm chart

**Prequisites for Monitoring Loki:**

> It is necessary to setup the Grafana Kubernetes Integration following the instructions in [Grafana Kubernetes Monitoring using Agent](/docs/grafana-cloud/kubernetes-monitoring/configuration/config-k8s-agent-guide/) as this will install necessary components for collecting metrics about your Kubernetes cluster and sending them to Grafana Cloud. Many of the dashboards installed as a part of the Loki integration rely on these metrics.

Reviewer 2: ```suggestion
You must setup the Grafana Kubernetes Integration following the instructions in [Grafana Kubernetes Monitoring using Agent](/docs/grafana-cloud/kubernetes-monitoring/configuration/config-k8s-agent-guide/) as this will install necessary components for collecting metrics about your Kubernetes cluster and sending them to Grafana Cloud. Many of the dashboards installed as a part of the Loki integration rely on these metrics.
```

Walking through this installation will create two Grafana Agent configurations, one for metrics and one for logs, that will add the external label `cluster: cloud`. In order for the Dashboards in the self-hosted Grafana Loki integration to work, the cluster name needs to match your Helm installation name. If you installed Loki using the command `helm install best-loki-cluster grafana/loki`, you would need to change the `cluster` value in both Grafana Agent configurations from `cloud` to `best-loki-cluster` when setting up the Grafana Kubernetes integration.

**To set up the Loki integration in Grafana Cloud:**

> 1. Get your Cloud Metrics and Cloud Logs credentials for your Grafana Cloud account.

Reviewer 3: I'd say that push permissions should suffice.

Author: :+1: changed

1. Create a secret in the same namespace as Loki to store your Cloud Logs credentials.

```bash
cat <<'EOF' | NAMESPACE=loki /bin/sh -c 'kubectl apply -n $NAMESPACE -f -'
apiVersion: v1
data:
  password: <BASE64_ENCODED_CLOUD_LOGS_PASSWORD>
  username: <BASE64_ENCODED_CLOUD_LOGS_USERNAME>
kind: Secret
metadata:
  name: grafana-cloud-logs-credentials
type: Opaque
EOF
```

> 1. Create a secret to store your Cloud Metrics credentials.

Author: yup, code block indentation was the problem. fixed.

```bash
cat <<'EOF' | NAMESPACE=loki /bin/sh -c 'kubectl apply -n $NAMESPACE -f -'
apiVersion: v1
data:
  password: <BASE64_ENCODED_CLOUD_METRICS_PASSWORD>
  username: <BASE64_ENCODED_CLOUD_METRICS_USERNAME>
kind: Secret
metadata:
  name: grafana-cloud-metrics-credentials
type: Opaque
EOF
```

> 1. Enable monitoring metrics and logs for the Loki installation to be sent your cloud database instances by adding the following to you Helm `values.yaml` file:

Reviewer: ```suggestion
1. Enable monitoring metrics and logs for the Loki installation to be sent your cloud database instances by adding the following to your Helm `values.yaml` file:
```

Reviewer 2: Numbering is off here in the preview. I think the code example needs to be indented.

    ```yaml
    ---
    monitoring:
      dashboards:

>         enabled: false

Author: The cloud integration will install the dashboards into the cloud grafana instance. This is disabling the dashaboards that would be installed into the same cluster that Loki is running by the helm charts. Those dashboards are not longer needed since the same dashboards will now be available in the cloud.

      rules:
        enabled: false
      selfMonitoring:
        logsInstance:
          clients:
            - url: <CLOUD_METRICS_URL>
              basicAuth:
                username:
                  name: grafana-cloud-logs-credentials
                  key: username
                password:
                  name: grafana-cloud-logs-credentials
                  key: password
      serviceMonitor:
        metricsInstance:
          remoteWrite:
            - url: <CLOUD_LOGS_URL>
              basicAuth:
                username:
                  name: grafana-cloud-metrics-credentials
                  key: username
                password:
                  name: grafana-cloud-metrics-credentials
                  key: password
    ```

> 1. Install the "Self-hosted Grafana Loki" intregation by going to your hosted Grafana instance, clicking the lightning bold icon labeled "Integrations and Connections", then search for and install the "Self-hosted Grafana Loki" integration.

Reviewer: ```suggestion
1. Install the "Self-hosted Grafana Loki" integration by going to your hosted Grafana instance, clicking the lightning bold icon labeled "Integrations and Connections", then search for and install the "Self-hosted Grafana Loki" integration.
```

> 1. Install the "Self-hosted Grafana Loki" intregation by going to your hosted Grafana instance, clicking the lightning bold icon labeled "Integrations and Connections", then search for and install the "Self-hosted Grafana Loki" integration.

Reviewer 2: ```suggestion
1. Install the self-hosted Grafana Loki integration by going to your hosted Grafana instance, clicking the lightning bolt icon labeled **Integrations and Connections**, then search for and install the **Self-hosted Grafana Loki** integration.
```

Author: fixed, thanks!

> 1. Once the "Self-hosted Grafana Loki" integration is installed, you should be able to clik the `View Dashboards` button to see the installed dashbards.

Reviewer: ```suggestion
1. Once the "Self-hosted Grafana Loki" integration is installed, you should be able to click the `View Dashboards` button to see the installed dashboards.
```

> 1. Once the "Self-hosted Grafana Loki" integration is installed, you should be able to clik the `View Dashboards` button to see the installed dashbards.

Reviewer 2: ```suggestion
1. Once the self-hosted Grafana Loki integration is installed, you should be able to clink the **View Dashboards** button to see the installed dashboards.
```

Author: fixed

> 1. Once the self-hosted Grafana Loki integration is installed, you should be able to click the **View Dashboards** button to see the installed dashboards.

Reviewer 2: ```suggestion
1. Once the self-hosted Grafana Loki integration is installed, click the **View Dashboards** button to see the installed dashboards.
```
